src/redis_lock/__init__.py

```diff
@@ -197,11 +231,13 @@ def __exit__(self, exc_type=None, exc_value=None, traceback=None, force=False):
         if self._lock_renewal_thread is not None:
             self._stop_lock_renewer()
         logger.debug("Releasing %r.", self._name)
-        try:
-            self._client.evalsha(UNLOCK_SCRIPT_HASH, 2, self._name, self._signal, self._id)
-        except NoScriptError:
+        is_cached, _ = _redis_eval(
```

Comment:
I think it would be much more nicer if we move all the scripts in a dict and `_redis_eval` also does the warning. Then `_redis_eval` would only take the "script_name" as argument and look it up in the dict (or dicts, if we have another dict with the hashes).

What do you think?